hawc/apps/common/management/commands/clean_doi.py

```diff
@@ -0,0 +1,158 @@
+import html
+import json
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from hawc.apps.lit import constants
+from hawc.apps.lit.models import Identifiers, Reference
+
+
+class Command(BaseCommand):
+    help = """Clean doi IDs and attempt to create doi ids for references without one"""
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--extensive",
+            action="store_true",
+            help="Attempt to extract DOI ids from all content fields",
+        )
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        doi_identifiers = Identifiers.objects.filter(database=4)
+        doi_id_initial = doi_identifiers.count()
+        validate_dois(doi_identifiers)
+        refs = Reference.objects.exclude(identifiers__database=4)
+        doi_ref_initial = Reference.objects.filter(identifiers__database=4).count()
+        if options["extensive"]:
+            create_dois_extensive(refs)
+        else:
+            create_dois(refs)
+        doi_id_final = Identifiers.objects.filter(database=4).count()
+        doi_ref_final = Reference.objects.filter(identifiers__database=4).count()
+        ref_total = Reference.objects.count()
+        print(f"Began with {doi_id_initial} doi IDs; finished with {doi_id_final} doi IDs")
+        print(
+            f"Began with {doi_ref_initial} references with doi IDs; finished with {doi_ref_final} references with doi IDs, out of {ref_total} total references"
+        )
+
+
+def validate_dois(doi_identifiers):
+    print(f"Validating {doi_identifiers.count()} doi IDs...")
+    doi_id_saved = {}
+    doi_id_updated = []
+    for doi_id in doi_identifiers:
+        if constants.DOI_EXACT.fullmatch(doi_id.unique_id):
+            doi_id_saved[doi_id.unique_id] = doi_id
+            # doi_id_saved.append(doi_id.unique_id)
+
+    duplicate_doi_ids = {}  # stores unclean DOIs that have a clean duplicate for updating later
+    tarnished_doi_ids = []  # tarnished DOI IDs cannot be cleaned/validated (ex: '10')
+    for doi_id in doi_identifiers:
```

Comment:
rename to `doi_ident`, or just `ident`? when i see `doi_id`, i think it's just a str or an id, not an object